### ISSUE TITLE
Compilation issues on macOS 

### ISSUE THREAD

**Issue by yashbhutwala** [citp/BlockSci#64]
Upon running make in libs/bitcoin-api-cpp, I get the following error: 

```
[ 50%] Built target bitcoinapi_static
[ 75%] Linking CXX shared library libbitcoinapi.dylib
Undefined symbols for architecture x86_64:
  "Json::ValueIterator::ValueIterator(Json::ValueIterator const&)", referenced from:
      BitcoinAPI::getaddednodeinfo(bool) in bitcoinapi.cpp.o
      BitcoinAPI::getaddednodeinfo(bool, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in bitcoinapi.cpp.o
      BitcoinAPI::getpeerinfo() in bitcoinapi.cpp.o
      BitcoinAPI::listreceivedbyaccount(int, bool) in bitcoinapi.cpp.o
      BitcoinAPI::listreceivedbyaddress(int, bool) in bitcoinapi.cpp.o
      BitcoinAPI::gettransaction(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, bool) in bitcoinapi.cpp.o
      BitcoinAPI::listtransactions() in bitcoinapi.cpp.o
      ...

__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) in bitcoinapi.cpp.o
      ...
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [src/bitcoinapi/libbitcoinapi.0.1.0.dylib] Error 1
make[1]: *** [src/bitcoinapi/CMakeFiles/bitcoinapi.dir/all] Error 2
make: *** [all] Error 2
```

**Comment by hkalodner** [comment]
Hi @yashbhutwala,

We don't currently provide official support for the mac (although I do test BlockSci on my personal apple laptop so I know it works).

I can definitely help walk you through the setup process. Most or all of the packages should be available from homebrew.

You current issue can be fixed with `brew install jsoncpp`.

Unfortunately there are a fair number of dependencies and it's difficult to get a clean system to test out fresh installs.

**Comment by yashbhutwala** [comment]
Thanks @hkalodner. I'm definitely learning that the local installation process is difficult to get right, which is why I was trying to get it right on my laptop first, and add the documentation in this repo for other developers. So far mirroring the Ubuntu dependency list, I've come up with the following brew equivalent for mac:

```
brew install cmake openssl boost sqlite3 autogen \
autoconf jsoncpp libjson-rpc-cpp \
python lmdb google-sparsehash argtable libmicrohttpd \
hiredis gcc gflags snappy \
lz4 zstd
```

And I did have jsoncpp installed when I received the above error.

**Comment by hkalodner** [comment]
Ah. I think this was actually a problem in the CMake script for bitcoin-api-cpp. I updated it in 3a4ffc4 in it should hopefully be working now.

**Comment by yashbhutwala** [comment]
Ok, sweet! 

P.S. notes on other minor hiccups along the way.

1) openssl is actually linked to LibreSSL on macOS High Sierra. So I had to run the BlockSci cmake as: 

`cmake -DOPENSSL_INCLUDE_DIR=/usr/bin/openssl -DCMAKE_BUILD_TYPE=Release ..`

2) Also, during make, I got the following error: 'openssl/sha.h' file not found. This was fixed by sym link as follows:
 
```
cd /usr/local/include 
ln -s ../opt/openssl/include/openssl .
```

Finally, I got it working locally tho. Thanks for the help, excited to play with this over the weekend!
